### PR TITLE
[7.4.0] Implement disk cache garbage collection.

### PR DESCRIPTION
**Pull request by tjgq**
By cherry-picking the following changes:

* 33aca3fe8803ca55838af5467fa801d03e28ffd8
* 85b100b93d8ff34f6ddc9e8530f468e61e0baf62
* 08cd046ee278d80d749ae20930ef439c2ad3c43b
* ca802d6bd704803e3861780bdb7f1ee50eca43d4
* 169006731e811e0542bc99287ca2ee7e150465c8
* 6096b5e49dbc4657f4447da44ca8f6fb3e6f9a8c
* 041faf5e884630ede2b7748463e0094366c2463a
* 46d5502745c97cb658349a0b0f4ddeef5f219057
* d48e3915ff292786b5a2196c8e296d04445a3745
* c43cf67b40a2646ff7c5443ef7d4e3562e06ac08
* db4a400806cee783cdcf8e4e67403f7a6ece028c
* 6922734a7bd6cc504cfc3390f4fa30e9438d1edd
* 544e23503e4c109335b12bfe9168a6a39c81c5b9
* b09cd65525f2e0a9bd56bc93e3f87bf36d91fd0b
* c72af54fcf692ae553071071adda9d29ffad3cdc
* 5997fdebe5c91e382148e67197d82e35c0034858
* caa2c413898e4353eaf3c31391576057a44f39aa

### CODE REVIEW

**Review by meteorcloudy — APPROVED** [review]
🎉

**Comment by luispadron** [comment]
Thanks ya'll for all the work on getting this into Bazel 7!

File: .bazelci/postsubmit.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ tasks:
       - "//src:bazel_jdk_minimal"
       - "//src:test_repos"
       - "//src/main/java/..."
+      - "//src/tools/diskcache/..."
```
Comment (Chino1307) [inline .bazelci/postsubmit.yml:21]: .bazelci/postsubmit.yml

**Review by Chino1307 — COMMENTED** [review]
.bazelci/postsubmit.yml

File: src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java [hunk 2]
```diff
@@ -64,6 +64,8 @@
 import com.google.devtools.build.lib.remote.circuitbreaker.CircuitBreakerFactory;
 import com.google.devtools.build.lib.remote.common.RemoteCacheClient;
 import com.google.devtools.build.lib.remote.common.RemoteExecutionClient;
+import com.google.devtools.build.lib.remote.disk.DiskCacheClient;
+import com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask;
```
Comment (Chino1307) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java:68]: src/BUILD

**Review by Chino1307 — COMMENTED** [review]
src/BUILD

File: .bazelci/presubmit.yml
Comment (Chino1307) [inline .bazelci/presubmit.yml:1]: .bazelci/presubmit.yml

File: src/main/java/com/google/devtools/build/lib/remote/disk/BUILD [hunk 3]
```diff
@@ -15,12 +15,17 @@ java_library(
     name = "disk",
     srcs = glob(["*.java"]),
     deps = [
+        "//src/main/java/com/google/devtools/build/lib/concurrent",
```
Comment (Chino1307) [inline src/main/java/com/google/devtools/build/lib/remote/disk/BUILD:18]: src/main/java/com/google/devtools/build/lib/remote/disk/BUILD

**Review by Chino1307 — COMMENTED** [review]
src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheClient.java

File: .bazelci/postsubmit.yml
Comment (Chino1307) [inline .bazelci/postsubmit.yml:1]: .bazelci/postsubmit.yml

File: src/main/java/com/google/devtools/build/lib/remote/disk/BUILD [hunk 4]
```diff
@@ -15,12 +15,17 @@ java_library(
...
         "//src/main/java/com/google/devtools/build/lib/remote/common:cache_not_found_exception",
+        "//src/main/java/com/google/devtools/build/lib/remote/options",
         "//src/main/java/com/google/devtools/build/lib/remote/util",
+        "//src/main/java/com/google/devtools/build/lib/server:idle_task",
+        "//src/main/java/com/google/devtools/build/lib/util:string",
         "//src/main/java/com/google/devtools/build/lib/vfs",
+        "//third_party:flogger",
         "//third_party:guava",
         "//third_party:jsr305",
         "//third_party/protobuf:protobuf_java",
```
Comment (Chino1307) [inline src/main/java/com/google/devtools/build/lib/remote/disk/BUILD:31]: .bazelci/postsubmit.yml

**Review by Chino1307 — COMMENTED** [review]
src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheClient.java

**Pull request merged**